### PR TITLE
Fix some comments for custom errors.

### PR DESCRIPTION
**Pull request by DaniOrtegaB**
Let's discuss here the new changes in comments and imports.

### CODE REVIEW

File: onchain/rollups/contracts/consensus/authority/Authority.sol [hunk 1]
```diff
@@ -43,6 +41,9 @@ contract Authority is AbstractConsensus, Ownable {
     /// @dev MUST be triggered on a successful call to `setHistory`.
     event NewHistory(IHistory history);
 
+    // @notice Raised when a transfer of tokens from an authority to a recipient fails.
```
Comment (guidanoli) [inline onchain/rollups/contracts/consensus/authority/Authority.sol:44]: Although errors don't carry NatSpec with them, it's good to prepend documentation comments with `///` (who knows if they make structs and errors support NatSpec in the future?).

Comment (DaniOrtegaB) [reply onchain/rollups/contracts/consensus/authority/Authority.sol]: Addressed in [69c6541](https://github.com/cartesi/rollups/commit/69c65413cc5636e9ec935e851bd549d85352da9b).

File: onchain/rollups/contracts/dapp/CartesiDApp.sol [hunk 2]
```diff
@@ -78,6 +74,15 @@ contract CartesiDApp is
     using Bitmask for mapping(uint256 => uint256);
     using LibOutputValidation for OutputValidityProof;
 
+    // @notice Raised when executing an already executed voucher.
```
Comment (guidanoli) [inline onchain/rollups/contracts/dapp/CartesiDApp.sol:77]: My previous comment applies to all custom errors, not just the one I'm highlighting with this very comment! :-)

File: onchain/rollups/contracts/library/LibOutputValidation.sol [hunk 3]
```diff
@@ -44,6 +39,22 @@ struct OutputValidityProof {
 library LibOutputValidation {
     using CanonicalMachine for CanonicalMachine.Log2Size;
 
+    /// @notice Raised when some `OutputvalidityProof` variables does not match
+    ///          the presented finalized epoch.
```
Comment (guidanoli) [inline onchain/rollups/contracts/library/LibOutputValidation.sol:43]: Remove extra space before "the".

File: onchain/rollups/contracts/library/LibOutputValidation.sol [hunk 4]
```diff
@@ -44,6 +39,22 @@ struct OutputValidityProof {
 library LibOutputValidation {
     using CanonicalMachine for CanonicalMachine.Log2Size;
 
+    /// @notice Raised when some `OutputvalidityProof` variables does not match
```
Comment (guidanoli) [inline onchain/rollups/contracts/library/LibOutputValidation.sol:42]: `OutputvalidityProof` ➡️ `OutputValidityProof`

This applies to several comments added by this PR.

File: onchain/rollups/contracts/library/LibOutputValidation.sol [hunk 5]
```diff
@@ -44,6 +39,22 @@ struct OutputValidityProof {
...
+    ///          the presented finalized epoch.
+    error IncorrectEpochHash();
+
+    /// @notice Raised when `OutputvalidityProof` metadata memory range is NOT
+    ///         contained in epoch's output memory range.
+    error IncorrectOutputsEpochRootHash();
+
+    /// @notice Raised when Merkle root hash of bytes(hashOfOutput) is NOT
```
Comment (guidanoli) [inline onchain/rollups/contracts/library/LibOutputValidation.sol:50]: For the user, we don't need to say we are casting `hashOfOutput` into `bytes`. This is just an internal limitation of our Merkle library that receives `bytes` and not `bytes32` to calculate the root hash. We could also just use the words "output hash" instead.

File: onchain/rollups/contracts/portals/IEtherPortal.sol [hunk 6]
```diff
@@ -26,7 +26,7 @@ interface IEtherPortal is IPortal {
     /// @param _dapp The address of the DApp
     /// @param _execLayerData Additional data to be interpreted by the execution layer
     /// @dev All the value sent through this function is forwarded to the DApp.
-    /// If the transfer fails, an `EtherTransferFailed` error is raised.
+    ///      If the transfer fails, `EtherTransferFailed` error is raised.
```
Comment (guidanoli) [inline onchain/rollups/contracts/portals/IEtherPortal.sol:29]: Why remove the indefinite article "an"?

**Review by guidanoli — COMMENTED** [review]
We should standardize the padding of spaces in `InputEncoding.sol`... How about we remove all extra spaces, and just pad the spaces with the longest line, like in the following example?

```solidity
abi.encodePacked(
    token, //   20B
    sender, //  20B
    tokenId, // 32B
    data //     arbitrary size
);
```

I would advise, however, that we leave this change to a later PR (as it has nothing to do with custom errors) on top of `main` and revert any related changes made in PR #49, so we can properly isolate this change in a commit with the `style` kind.

**Comment by DaniOrtegaB** [comment]
> We should standardize the padding of spaces in `InputEncoding.sol`... How about we remove all extra spaces, and just pad the spaces with the longest line, like in the following example?
> 
> ```solidity
> abi.encodePacked(
>     token, //   20B
>     sender, //  20B
>     tokenId, // 32B
>     data //     arbitrary size
> );
> ```
> 
> I would advise, however, that we leave this change to a later PR (as it has nothing to do with custom errors) on top of `main` and revert any related changes made in PR #49, so we can properly isolate this change in a commit with the `style` kind.

I agree on modifying the padding later, but for this PR what we have is all paddings at the same level, maybe we should change how much padding but right now is slightly better than it was, as not it's more consistent.

File: onchain/rollups/contracts/library/LibOutputValidation.sol (hunk 5 above)
Comment (DaniOrtegaB) [reply onchain/rollups/contracts/library/LibOutputValidation.sol:50]: Totally agree, I'll also remove `hash` from `Merkle root hash` as it's redundant, a merkle root itself it a hash.

**Comment by DaniOrtegaB** [comment]
Rebased on top of `feature/custom-errors`.

**Comment by guidanoli** [comment]
> I agree on modifying the padding later, but for this PR what we have is all paddings at the same level, maybe we should change how much padding but right now is slightly better than it was, as not it's more consistent.

Yes, but since this change is unrelated to custom errors, and so I think it deserves a PR of its own.

File: onchain/rollups/contracts/library/LibOutputValidation.sol (hunk 5 above)
Comment (guidanoli) [reply onchain/rollups/contracts/library/LibOutputValidation.sol]: Yes, it is redundant, but we use this terminology all over the documentation.
So, unless we're adopting this nomenclature everywhere, I'd rather keep the redundancy.

**Comment by DaniOrtegaB** [comment]
Agree totally, but undo a better comment than previously? With the new PR we'll adjust the padding.

File: onchain/rollups/contracts/portals/IEtherPortal.sol (hunk 6 above)
Comment (guidanoli) [reply onchain/rollups/contracts/portals/IEtherPortal.sol:29]: Well, that is half true. You could think of error types like classes. A class can be instantiated multiple times. So, you'd say that a function returns an instance, and not the class itself. Likewise, a function may raise an error of a specific type, but not the error type itself.

**Comment by guidanoli** [comment]
> Agree totally, but undo a better comment than previously? With the new PR we'll adjust the padding.

Yes, I think it makes sense to change the comments since they are related to custom errors. We'll leave the padding to a future PR.

**Comment by guidanoli** [comment]
I would revert the change in the comment inside the `LibOutputValidation`, removing the word "hash", however, as it is not related to custom errors.

File: onchain/rollups/contracts/portals/EtherPortal.sol [hunk 7]
```diff
@@ -17,13 +17,14 @@ import {Portal} from "./Portal.sol";
...
 import {InputEncoding} from "../common/InputEncoding.sol";
 
-error EtherTransferFailed();
-
 /// @title Ether Portal
 ///
 /// @notice This contract allows anyone to perform transfers of
 /// Ether to a DApp while informing the off-chain machine.
 contract EtherPortal is Portal, IEtherPortal {
+    /// @notice Raised when the Ether the transfer fails.
```
Comment (ZzzzHui) [inline onchain/rollups/contracts/portals/EtherPortal.sol:25]: typo here: ether transfer

**Review by ZzzzHui — APPROVED** [review]

**Pull request merged**